src/main/java/controllers/ProjectInputController.java

```diff
@@ -0,0 +1,309 @@
+package controllers;
+
+import models.data.Project;
+import models.member.IMember;
+import models.member.Member;
+import models.task.ITask;
+import models.task.Task;
+import repositories.ProjectRepository;
+import util.factories.MemberFactory;
+import util.factories.TaskFactory;
+import util.log.DukeLogger;
+import views.CLIView;
+
+import java.text.ParseException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public class ProjectInputController implements IController {
+    private Scanner manageProjectInput;
+    private ProjectRepository projectRepository;
+    private CLIView consoleView;
+    private MemberFactory memberFactory;
+
+    /**
+     * Constructor for ProjectInputController takes in a View model and a ProjectRepository.
+     * ProjectInputController is responsible for handling user input when user chooses to manage a project.
+     * @param consoleView The main UI of ArchDuke.
+     * @param projectRepository The object holding all projects.
+     */
+    public ProjectInputController(CLIView consoleView, ProjectRepository projectRepository) {
+        this.manageProjectInput = new Scanner(System.in);
+        this.projectRepository = projectRepository;
+        this.consoleView = consoleView;
+        this.memberFactory = new MemberFactory();
+    }
+
+    /**
+     * Allows the user to manage the project by branching into the project of their choice.
+     * @param input User input containing project index number (to add to project class).
+     */
+    public void onCommandReceived(String input) {
+        DukeLogger.logInfo(ProjectInputController.class, "Managing project: " + input);
+        int projectNumber = Integer.parseInt(input);
+        Project projectToManage = projectRepository.getItem(projectNumber);
+        this.consoleView.consolePrint("Now managing: " + projectToManage.getDescription());
+        boolean isManagingAProject = true;
+        while (isManagingAProject) {
+            isManagingAProject = manageProject(projectToManage);
+        }
+    }
+
+    /**
+     * Manages the project.
+     * @param projectToManage The project specified by the user.
+     * @return Boolean variable giving status of whether the exit command is entered.
+     */
+    public boolean manageProject(Project projectToManage) {
+        boolean isManagingAProject = true;
+        if (manageProjectInput.hasNextLine()) {
+            String projectCommand = manageProjectInput.nextLine();
+            DukeLogger.logInfo(ProjectInputController.class, "Managing:"
+                    + projectToManage.getDescription() + ",input:'"
+                    + projectCommand + "'");
+            if (projectCommand.length() == 4 && ("exit").equals(projectCommand.substring(0, 4))) {
```

Comment:
Any particular reason why we're checking for command length?

Reply:
Understooded